app/forms/refunds/fees_form.rb

```diff
@@ -0,0 +1,150 @@
+
+    validates :et_reconsideration_fee_payment_method,
+      presence: true,
+      inclusion: PAYMENT_METHODS,
+      if: -> { et_reconsideration_fee.present? && et_reconsideration_fee.positive? }
+
+    validates :et_reconsideration_fee_payment_date,
+      presence: true,
+      date: true,
+      if: lambda {
+            et_reconsideration_fee.present? &&
+              et_reconsideration_fee.positive? &&
+              !et_reconsideration_fee_payment_date_unknown?
+          }
+
+    validates :eat_issue_fee_payment_method,
+      presence: true,
+      inclusion: PAYMENT_METHODS,
+      if: lambda {
+        eat_issue_fee.present? && eat_issue_fee.positive?
+      }
+
+    validates :eat_issue_fee_payment_date, presence: true, date: true,
+                                           if: lambda {
+                                                 eat_issue_fee.present? &&
+                                                   eat_issue_fee.positive? &&
+                                                   !eat_issue_fee_payment_date_unknown?
+                                               }
+
+    validates :eat_hearing_fee_payment_method,
+      presence: true,
+      inclusion: PAYMENT_METHODS,
+      if: lambda {
+        eat_hearing_fee.present? && eat_hearing_fee.positive?
+      }
+
+    validates :eat_hearing_fee_payment_date, presence: true, date: true,
+                                             if: lambda {
+                                                   eat_hearing_fee.present? &&
+                                                     eat_hearing_fee.positive? &&
+                                                     !eat_hearing_fee_payment_date_unknown?
+                                                 }
+
+    def et_issue_fee_payment_date_with_pre_process=(val)
+      self.et_issue_fee_payment_date_without_pre_process = pre_process_partial_date(val, only: :day)
+    end
+    alias_method_chain :et_issue_fee_payment_date=, :pre_process
+
+    def et_hearing_fee_payment_date_with_pre_process=(val)
+      self.et_hearing_fee_payment_date_without_pre_process = pre_process_partial_date(val, only: :day)
+    end
+    alias_method_chain :et_hearing_fee_payment_date=, :pre_process
+
+    def et_reconsideration_fee_payment_date_with_pre_process=(val)
+      self.et_reconsideration_fee_payment_date_without_pre_process = pre_process_partial_date(val, only: :day)
+    end
+    alias_method_chain :et_reconsideration_fee_payment_date=, :pre_process
+
+    def eat_issue_fee_payment_date_with_pre_process=(val)
+      self.eat_issue_fee_payment_date_without_pre_process = pre_process_partial_date(val, only: :day)
+    end
+    alias_method_chain :eat_issue_fee_payment_date=, :pre_process
+
+    def eat_hearing_fee_payment_date_with_pre_process=(val)
+      self.eat_hearing_fee_payment_date_without_pre_process = pre_process_partial_date(val, only: :day)
+    end
+    alias_method_chain :eat_hearing_fee_payment_date=, :pre_process
+
+    private
```

Comment:
The headache is that there is a class method called 'dates' that does some weird stuff that looked ugly.  Hence, I could not use super here or anything - so had to use alias method chain which I hated, but the form stuff is not very nice code and does some over complex stuff.  I would prefer to spend the time upgrading to rails 5 and removing these awful form objects anyway - there are so many workarounds in the code because of it.

We could do a concern that does the alias method chain etc.. but the problem is, it would need to be included after the 'dates' class method call otherwise it would not work.

I tried really hard to keep this area clean, but the form object code was just causing issues all the time and this was the best I could come up with

Reply:
Ive now put the 'dates' class method call and the date transformation code into a concern to hide this complexity.